libs/gen/EnumsSourceGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Garnet;
+
+[Generator]
+public class EnumsSourceGenerator : IIncrementalGenerator
+{
+    const string GeneratedClassName = "EnumUtils";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var enumDetails = context.SyntaxProvider
+            .CreateSyntaxProvider(
```

Comment:
We should use [`ForAttributeWithMetadataName`](https://andrewlock.net/creating-a-source-generator-part-9-avoiding-performance-pitfalls-in-incremental-generators/#1-use-the-net-7-api-forattributewithmetadataname) which is specially optimized in the Roslyn.